pkg/apis/core/validation/validation.go

```diff
@@ -1243,13 +1243,33 @@ func validateMountPropagation(mountPropagation *core.MountPropagationMode, conta
 		return allErrs
 	}
 
-	privileged := container.SecurityContext != nil && container.SecurityContext.Privileged != nil && *container.SecurityContext.Privileged
-	if *mountPropagation == core.MountPropagationBidirectional && !privileged {
-		allErrs = append(allErrs, field.Forbidden(fldPath, "Bidirectional mount propagation is available only to privileged containers"))
+	if *mountPropagation == core.MountPropagationBidirectional {
+		// additional permissions required for mount functionality
+		if !hasBidirectionalMountPropagationPermissions(container.SecurityContext) {
+			allErrs = append(allErrs, field.Forbidden(fldPath, "Bidirectional mount propagation is available only to privileged containers and containers with the 'SYS_ADMIN' capability"))
+		}
 	}
 	return allErrs
 }
 
+func hasBidirectionalMountPropagationPermissions(securityContext *core.SecurityContext) bool {
+	if securityContext == nil {
+		// No security context configured
+		return false
+	}
+
+	// check container for required capability
+	if securityContext.Capabilities != nil {
+		for _, capability := range securityContext.Capabilities.Add {
+			if capability == "SYS_ADMIN" {
+				return true
+			}
+		}
+	}
+	// check if container is privileged
+	return securityContext.Privileged != nil && *securityContext.Privileged
```

Comment:
do this check first since it's cheaper than the capabilities check, and since we'll want to short-circuit if the `opts.AllowUnprivilegedBidirectionalMountPropagation` flag is false

Reply:
Split check into:
1. privileged check
2. unprivileged flag + capabilities check